packages/blocks/src/store/reducer.js

```diff
@@ -124,10 +134,27 @@ export const unregisteredFallbackBlockName = createBlockNameSetterReducer( 'SET_
  * @return {Object} Updated state.
  */
 export function categories( state = DEFAULT_CATEGORIES, action ) {
-	if ( action.type === 'SET_CATEGORIES' ) {
-		return action.categories || [];
+	switch ( action.type ) {
+		case 'SET_CATEGORIES':
```

Comment:
Should categories be stored in an object keyed by the slug? It would simplify update operations. It might negatively influence selectors though.